### ISSUE TITLE
wav2vec 2.0 finetuning problem with padding mask

### ISSUE THREAD

**Issue by joonson** [pytorch/fairseq#3573]
I have the following error when fine-tuning a pre-trained wav2vec 2.0 model. Anyone with similar problems?

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
    fn(i, *args)
  File "/fairseq/fairseq/distributed/utils.py", line 328, in distributed_main
    main(cfg, **kwargs)
  File "/fairseq/fairseq_cli/train.py", line 178, in main
    valid_losses, should_stop = train(cfg, trainer, task, epoch_itr)
  File "/usr/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/fairseq/fairseq_cli/train.py", line 289, in train
    log_output = trainer.train_step(samples)
  File "/usr/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/fairseq/fairseq/trainer.py", line 705, in train_step
    ignore_grad=is_dummy_batch,
  File "/fairseq/fairseq/tasks/fairseq_task.py", line 475, in train_step
    loss, sample_size, logging_output = criterion(model, sample)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/fairseq/fairseq/criterions/ctc.py", line 114, in forward
    non_padding_mask = ~net_output["padding_mask"]
TypeError: bad operand type for unary ~: 'NoneType'
```

Default settings (below) was used:
```
fairseq-hydra-train \
    distributed_training.distributed_port=$PORT \
    task.data=/path/to/data \
    model.w2v_path=/path/to/model.pt \
    --config-dir /path/to/fairseq-py/examples/wav2vec/config/finetuning \
    --config-name base_100h
```

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0) 1.8.1
 - OS (e.g., Linux): linux
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source): `pip install --editable ./`
 - Python version: 3.6.9
 - CUDA/cuDNN version: 11.1
 - GPU models and configuration: P40 x 8
 - Any other relevant information:

**Comment by innarid** [comment]
Same issue

**Comment by alexeib** [comment]
fixed in e6eddd805ebbc5c17bf5100c2fde6e0dfc946d2c

**Comment by jinhao2557** [comment]
> I have the following error when fine-tuning a pre-trained wav2vec 2.0 model. Anyone with similar problems?
> 
> ```
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/dist-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
>     fn(i, *args)
>   File "/fairseq/fairseq/distributed/utils.py", line 328, in distributed_main
>     main(cfg, **kwargs)
>   File "/fairseq/fairseq_cli/train.py", line 178, in main
>     valid_losses, should_stop = train(cfg, trainer, task, epoch_itr)
>   File "/usr/lib/python3.6/contextlib.py", line 52, in inner
>     return func(*args, **kwds)
>   File "/fairseq/fairseq_cli/train.py", line 289, in train
>     log_output = trainer.train_step(samples)
>   File "/usr/lib/python3.6/contextlib.py", line 52, in inner
>     return func(*args, **kwds)
>   File "/fairseq/fairseq/trainer.py", line 705, in train_step
>     ignore_grad=is_dummy_batch,
>   File "/fairseq/fairseq/tasks/fairseq_task.py", line 475, in train_step
>     loss, sample_size, logging_output = criterion(model, sample)
>   File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 889, in _call_impl
>     result = self.forward(*input, **kwargs)
>   File "/fairseq/fairseq/criterions/ctc.py", line 114, in forward
>     non_padding_mask = ~net_output["padding_mask"]
> TypeError: bad operand type for unary ~: 'NoneType'
> ```
> 
> Default settings (below) was used:
> 
> ```
> fairseq-hydra-train \
>     distributed_training.distributed_port=$PORT \
>     task.data=/path/to/data \
>     model.w2v_path=/path/to/model.pt \
>     --config-dir /path/to/fairseq-py/examples/wav2vec/config/finetuning \
>     --config-name base_100h
> ```
> 
> #### What's your environment?
> 
>     * fairseq Version (e.g., 1.0 or master): master
> 
>     * PyTorch Version (e.g., 1.0) 1.8.1
> 
>     * OS (e.g., Linux): linux
> 
>     * How you installed fairseq (`pip`, source): pip
> 
>     * Build command you used (if compiling from source): `pip install --editable ./`
> 
>     * Python version: 3.6.9
> 
>     * CUDA/cuDNN version: 11.1
> 
>     * GPU models and configuration: P40 x 8
> 
>     * Any other relevant information:



> I have the following error when fine-tuning a pre-trained wav2vec 2.0 model. Anyone with similar problems?
> 
> ```
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/dist-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
>     fn(i, *args)
>   File "/fairseq/fairseq/distributed/utils.py", line 328, in distributed_main
>     main(cfg, **kwargs)
>   File "/fairseq/fairseq_cli/train.py", line 178, in main
>     valid_losses, should_stop = train(cfg, trainer, task, epoch_itr)
>   File "/usr/lib/python3.6/contextlib.py", line 52, in inner
>     return func(*args, **kwds)
>   File "/fairseq/fairseq_cli/train.py", line 289, in train
>     log_output = trainer.train_step(samples)
>   File "/usr/lib/python3.6/contextlib.py", line 52, in inner
>     return func(*args, **kwds)
>   File "/fairseq/fairseq/trainer.py", line 705, in train_step
>     ignore_grad=is_dummy_batch,
>   File "/fairseq/fairseq/tasks/fairseq_task.py", line 475, in train_step
>     loss, sample_size, logging_output = criterion(model, sample)
>   File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 889, in _call_impl
>     result = self.forward(*input, **kwargs)
>   File "/fairseq/fairseq/criterions/ctc.py", line 114, in forward
>     non_padding_mask = ~net_output["padding_mask"]
> TypeError: bad operand type for unary ~: 'NoneType'
> ```
> 
> Default settings (below) was used:
> 
> ```
> fairseq-hydra-train \
>     distributed_training.distributed_port=$PORT \
>     task.data=/path/to/data \
>     model.w2v_path=/path/to/model.pt \
>     --config-dir /path/to/fairseq-py/examples/wav2vec/config/finetuning \
>     --config-name base_100h
> ```
> 
> #### What's your environment?
> 
>     * fairseq Version (e.g., 1.0 or master): master
> 
>     * PyTorch Version (e.g., 1.0) 1.8.1
> 
>     * OS (e.g., Linux): linux
> 
>     * How you installed fairseq (`pip`, source): pip
> 
>     * Build command you used (if compiling from source): `pip install --editable ./`
> 
>     * Python version: 3.6.9
> 
>     * CUDA/cuDNN version: 11.1
> 
>     * GPU models and configuration: P40 x 8
> 
>     * Any other relevant information:

Have you solved this problem? Please help me. Thanks.
